### PR TITLE
Adding callback calls as a new Persistence behaviour

### PR DESCRIPTION
**Pull request by rmarinleal**
* Improving the SDK Persistent feature to allow a new way to get/set the persistent data supported by the use of callbacks.

### CODE REVIEW

**Review by ScottAntonacAP — APPROVED** [review]

File: src/PersistentStorage.php [hunk 1]
```diff
@@ -94,53 +94,71 @@ public static function testConnection()
...
-        } else {
-            throw new Exception("No available database API");
+        } elseif (isset($this->callback_active) && $this->callback_active === true) {
+            $this->callback_getLastUpdateDate = Config::get('callback.getLastUpdateDate');
+            $this->callback_setLastUpdateDate = Config::get('callback.setLastUpdateDate');
+            $this->callback_getOrderMinimum = Config::get('callback.getOrderMinimum');
+            $this->callback_setOrderMinimum = Config::get('callback.setOrderMinimum');
+            $this->callback_getOrderMaximum = Config::get('callback.getOrderMaximum');
+            $this->callback_setOrderMaximum = Config::get('callback.setOrderMaximum');
+            $this->callback_getCurrency = Config::get('callback.getCurrency');
```
[the remark this answers is not in the document]
Comment (benfterpay) [reply src/PersistentStorage.php:161]: In any case I think the config names are fine (e.g. `callback.getLastUpdateDate`). We could change how this works under the hood at any time, and as long as we don't change those config names or corresponding method arguments, we wouldn't break any implementations.

**Review by benfterpay — CHANGES_REQUESTED** [review]
In principle, I think this is a great way to leverage an application's existing persistent storage service from the SDK. The logic of whether to call an API or use locally cached data remains with the SDK, while how and where the data is persisted is transitioned to the application. Great idea. Removes the strict dependency on the mysqli driver and the configuration of connection credentials.

Just a couple of suggestions - some to be addressed now, but most could be addressed later (or never), depending on how this feature is implemented and the feedback received from those implementations.

File: src/PersistentStorage.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (rmarinleal) [reply src/PersistentStorage.php:169]: I've defined the variables above, I followed the same principle as the DB persistence do, and use top-level variables per each config defined on the Config.php file.

File: src/PersistentStorage.php [hunk 2]
```diff
@@ -94,53 +94,71 @@ public static function testConnection()
...
-        $this->db_tablePrefix = Config::get('db.tablePrefix');
-        $this->db_user = Config::get('db.user');
-        $this->db_pass = Config::get('db.pass');
-
-        if ($this->db_api == 'mysqli') {
-            if (extension_loaded('mysqli')) {
-                /**
-                 * @see https://www.php.net/manual/en/mysqli.construct.php
-                 */
+        $this->callback_active = Config::get('callback.active');
```
[the remark this answers is not in the document]
Comment (rmarinleal) [reply src/PersistentStorage.php:106]: Fixed, I've declared the variables on the same block than the DB ones.

**Review by benfterpay — APPROVED** [review]

**Pull request merged**